channels/applicationinsights-channel-js/src/Sender.ts

```diff
@@ -613,12 +639,99 @@ export class Sender extends BaseTelemetryPlugin implements IChannelControls {
                         _self._onError(payload, _formatErrorMessageXdr(xdr));
                     }
                 }
-            };
+
+            }
         
             function _isSampledIn(envelope: ITelemetryItem): boolean {
                 return _self._sample.isSampledIn(envelope);
             }
 
+            function _getSenderInterface(transports: TransportType[], syncSupport: boolean): IXHROverride {
+                let transportType: TransportType = null;
+                let sendPostFunc: SendPOSTFunction = null;
+                let lp = 0;
+                while (sendPostFunc == null && lp < transports.length) {
+                    transportType = transports[lp];
+                    if (!_disableXhr && transportType === TransportType.Xhr) {
+                        if (useXDomainRequest()) {
+                            // IE 8 and 9
+                            sendPostFunc = _xdrSender;
+                        } else if (isXhrSupported()) {
+                            sendPostFunc = _xhrSender;
+                        }
+                    } else if (transportType === TransportType.Fetch && isFetchSupported(syncSupport)) {
+                        sendPostFunc = _fetchSender;
+                    } else if (isBeaconsSupported() && transportType === TransportType.Beacon) {
+                        sendPostFunc = _beaconSender;
+                    }
+
+                    lp++;
+                }
+
+                if (sendPostFunc) {
+                    return {
+                        sendPOST: sendPostFunc
+                    };
+                }
+
+                return null;
+            }
+
+            function _getOnComplete(payload: string[], status: number, headers: {[headerName: string]: string;}, response?: string) {
+
+                // ***********************************************************************************************
+                //Note: status code should be discussed
+                if (status === 200) {
+                    _self._onSuccess(payload, payload.length);
+                }
+                if (status === 400) {
+                    _self._onError(payload, response);
+                }
+            }
+
+            function _getSender(sendInterface: IXHROverride, payload: string[], isAsync: boolean): void | IPromise<boolean> {
```

Comment:
This looks like it's actually "sending" (calling the sendPost) the data, so `_getSender` feels like the wrong name, perhaps `_doSend` or something like that?